### DISCUSSION TITLE
My text is being drawn offscreen even when drawing at 0,0. Why is this?

### DISCUSSION THREAD

**Discussion by AngryCarrot789** [mono/SkiaSharp#2735] [Q&A]
I'm trying to draw text so that all of the text contents are drawn at 0,0 relative to the top left corner, but for some reason all of the text is drawn far above that point. What causes this? This is the code I'm using:
```
using (SKPaint paint = new SKPaint() { IsAntialias = true, Color = SKColors.White }) {
    using (SKTextBlob blob = SKTextBlob.Create(text, fd.cachedFont)) {
        rc.Canvas.DrawText(blob, 0, 0, paint);
    }
}
```

**Comment by AngryCarrot789** [comment]
I drew a border around the outside of the text using `blob.Bounds`, and after translating the canvas matrix a bit, I get this:
![image](https://github.com/mono/SkiaSharp/assets/32274404/c7e7867d-0c93-480a-8481-9a0a07a5ea81)

I don't really understand why the blob's Bounds' Left and Top are in the negative and why there's so much padding on the Bounds

**Comment by taublast** [comment]
Hi, text is drawn not like an image, but is rendered using front metrics from the lower left corner. That's why if you draw at 0,0 the text will be just outside the canvas bounds.
You can checkout an example of a control drawing complex text
https://github.com/taublast/DrawnUi.Maui/tree/main/src/Engine/Draw/Text

**Reply by AngryCarrot789** [reply]
Yeah, I figured out it was drawing at the left side of the baseline, but I did manage to figure out [a rough workaround](https://github.com/AngryCarrot789/FramePFX/blob/master/FramePFX/Editors/Timelines/Clips/Core/TimecodeClip.cs#L145) (works for basic text)
